aws/stack/register_middleware.go

```diff
@@ -12,26 +12,30 @@ type AwsRetryMiddlewareConfig interface {
 	GetRetryer() retry.Retryer
 }
 
+// AwsRetryMiddlewares represents the Aws middleware's for Retry
+type AwsRetryMiddlewares struct {
+	*retry.AttemptMiddleware
+	*retry.MetricsHeaderMiddleware
+}
```

Comment:
probably want these as named fields instead of anonymous. You'll get unexpected results if either structure has clashing members.